packages/beacon-node/src/network/reqresp/handlers/dataColumnSidecarsByRange.ts

```
@@ -5,16 +5,22 @@ import {fulu} from "@lodestar/types";
 import {fromHex} from "@lodestar/utils";
 import {IBeaconChain} from "../../../chain/index.js";
 import {IBeaconDb} from "../../../db/index.js";
+import {filterDataColumnForCustody, logDataColumnSidecarUnavailability} from "../utils/dataColumnResponseValidaiton.js";
 
 export async function* onDataColumnSidecarsByRange(
   request: fulu.DataColumnSidecarsByRangeRequest,
   chain: IBeaconChain,
   db: IBeaconDb
 ): AsyncIterable<ResponseOutgoing> {
   // Non-finalized range of columns
-  const {startSlot, count, columns} = validateDataColumnSidecarsByRangeRequest(request);
+  const {startSlot, count, columns: requestedColumns} = validateDataColumnSidecarsByRangeRequest(request);
   const endSlot = startSlot + count;
 
+  const columns = filterDataColumnForCustody(requestedColumns, chain.custodyConfig.custodyColumns, chain.logger);
```

I think if peer requested a column we don't have, we should throw error instead of returning available column data
leaving here for @matthewkeil to confirm

---

As of my understanding of the specs, we are supposed to serve the data which we have and skip which we don't.

---

> we are supposed to serve the data which we have and skip which we don't.

this is correct

---

hmmm.  the spec is not very clear on this. if a peer requests something that they know we do not have (not part of our announced custody) i think we should throw a malformed request error.  If we do not have data that we should have we are supposed to return `3: Resource Unavailable` so if its something we know we dont have we should error out early